### ISSUE TITLE
[ISSUE] identity cannot be accessed under a sub-path

### ISSUE THREAD

**Issue by linonetwo** [camunda/camunda-platform-helm#1110]
**Describe the issue:**

Same as https://github.com/camunda/camunda-platform-helm/issues/479 , but solution in that issue is probably not correct, when accessing http://camunda.example.com/ , it will redirect to http://camunda.example.com//auth/login (two slash), and showing a 404 not found page of keycloak

after fixing "identity cannot be accessed under a sub-path", I get 404

![截屏2023-12-05 17 50 28](https://github.com/camunda/camunda-platform-helm/assets/3746270/09cff54b-53d2-4d35-8c7c-9f7346cfab9b)

**Expected behavior:**

Install chart and keycloak, and access identity service in / path

**How to reproduce:**

I try to combine https://github.com/camunda/camunda-platform-helm/blob/main/kind/camunda-platform-core-kind-values.yaml and https://docs.camunda.io/docs/self-managed/platform-deployment/helm-kubernetes/guides/ingress-setup/

for my microk8s single node cluster.

diff here

```yaml
global:
  ingress:
    enabled: true
    host: "my.domain"
  identity:
    auth:
      publicIssuerUrl: "http://my.domain/auth/realms/camunda-platform"
      operate:
        redirectUrl: "http://my.domain/operate"
      tasklist:
        redirectUrl: "http://my.domain/tasklist"
      optimize:
        redirectUrl: "http://my.domain/optimize"
# Reduce for Zeebe and Gateway the configured replicas and with that the required resources
# to get it running locally
zeebe:
  clusterSize: 1
  partitionCount: 1
  replicationFactor: 1
zeebe-gateway:
  replicas: 1
  ingress:
    enabled: true
    className: nginx
    host: "zeebe.my.domain"

# Configure elastic search to make it running for local development
elasticsearch:
  replicas: 1
  
identity:
  contextPath: "/"
  fullURL: "http://my.domain/"

operate:
  contextPath: "/operate"

optimize:
  contextPath: "/optimize"

tasklist:
  contextPath: "/tasklist"
```

**Logs:**

**Environment:**

**Please note: Without the following info, it's hard to resolve the issue and probably it will be closed.**

- Platform: SelfHost
- Helm CLI version: 3 (kubeapps)
- Chart version: 8.3.2
- Values file: 
[values.yml.zip](https://github.com/camunda/camunda-platform-helm/files/13558376/values.yml.zip)

```[tasklist]
### Tasks
```

**Comment by aabouzaid** [comment]
@linonetwo Identity doesn't work under a sub-path if no TLS is enabled. Either set it with its own Ingress domain to work with HTTP or configure TLS to make it work as a sub-path. 
https://docs.camunda.io/docs/next/self-managed/platform-deployment/troubleshooting/#identity-contextpath

**Comment by linonetwo** [comment]
tasklist app give correct redirect to http://my.domain/auth/realms/camunda-platform/protocol/openid-connect/auth?client_id=tasklist&redirect_uri=http%3A%2F%2Fmy.domain%2Ftasklist%2Fidentity-callback&response_type=code&scope=openid+email&state=

just identity doesn't. And due to this I can't evaluate the zeebe . I just want to connect to zeebe
<img width="308" alt="截屏2023-12-06 17 17 47" src="https://github.com/camunda/camunda-platform-helm/assets/3746270/d13b81df-984c-45f0-b7c9-2aa5cc125592">

**Comment by linonetwo** [comment]
I try to setup tls to avoid this issue, but 

```yaml
spec:
  ingressClassName: nginx
  tls:
    - hosts:
        - my.domain
      secretName: camunda-platform
```

don't seem to work

**Comment by cr1st1p** [comment]
a) identity.fullURL probably does not have to end up in a '/' - to avoid double // in the redirect url (I did not test with an ending '/')
b) to make it work, I had to also manually edit the main ingress to add /auth/login -> identity server, BEFORE the /auth ->keycloak entry

**Comment by korncola** [comment]
why is this issue closed? I ran exactly into the same issue! And there is no documentation about this in the values.yaml or the docs — the templates for ingress needs to be fixed to reflect @cr1st1p solution.

But even after editing the ingress, something with the callback is still wrong. I can reach the login window of identity, but after login I get the same error "We are sorry, page not found".

@cr1st1p did you had the same error.

Litte rant, sorry for that: I am baffled how anybody pays money for this duct tape pieced together overcomplicated legacy java thing, luckily I am doing this only as PoC to show my employer how bad Camunda is, which is easy to prove actually, cause we had so many issues setting it up.

**Comment by cr1st1p** [comment]
> But even after editing the ingress, something with the callback is still wrong. I can reach the login window of identity, but after login I get the same error "We are sorry, page not found".
> 
> [@cr1st1p](https://github.com/cr1st1p) did you had the same error?

Yes, I had, and I had to add like you an ingress rule for /auth/login and /auth/login-callback. I edited helm's ingress on the fly instead of adding a new ingress rule (made possible by Pulumi's transformations). I don't remember why I did not try adding a separate ingress rule.

I had to also transform 2 secrets, otherwise with every Pulumi run (argocd/fluxcd would most probably behave the same), I would have changes because those secrets were always random, without a way to overwrite.

Recently I had to add an ingress annotation, to cope with errors from nginx >= 1.13 regarding proxy-busy-buffers-size: "nginx.ingress.kubernetes.io/proxy-busy-buffers-size": "128k"

Hope that helps. Regards.
